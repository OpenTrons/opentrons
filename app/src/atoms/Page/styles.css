.page {
  width: 100%;
  height: 100vh;
  overflow-y: auto;
}

.sticky_title_bar {
  position: sticky;
  top: 0;
  min-height: 3rem;
  width: 100%;
  z-index: 1;
}
