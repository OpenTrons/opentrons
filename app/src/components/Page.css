.task {
  flex: 1 1;
  position: relative;
}
