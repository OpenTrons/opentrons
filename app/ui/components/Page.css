.task {
  flex: 1 1;
  border-left: 1px solid silver;
  position: relative;
}
