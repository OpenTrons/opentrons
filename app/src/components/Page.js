// task component
import React from 'react'

import styles from './Page.css'
import LostConnectionAlert from './LostConnectionAlert'
import {AnalyticsSettingsModal} from './analytics-settings'

export default function Page (props) {
  return (
    <main className={styles.task}>
      {props.children}
      <LostConnectionAlert />
      <AnalyticsSettingsModal />
    </main>
  )
}
