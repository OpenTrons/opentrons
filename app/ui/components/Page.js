// task component
import React from 'react'
import styles from './Page.css'

export default function Page (props) {
  return (
    <main className={styles.task}>
      {props.children}
    </main>
  )
}
