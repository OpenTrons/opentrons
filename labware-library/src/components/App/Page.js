// @flow
import * as React from 'react'
import { Switch, Route } from 'react-router-dom'

import { getDefinition } from '../../definitions'
import { getFilters } from '../../filters'
import { getPublicPath } from '../../public-path'
import Sidebar from '../Sidebar'
import LabwareList, { NoResults } from '../LabwareList'
import LabwareDetails from '../LabwareDetails'
import styles from './styles.css'

import type { Location } from 'react-router-dom'

export type PageProps = {
  location: Location,
}

export default function Page(props: PageProps) {
  const filters = getFilters(props.location)

  return (
    <div className={styles.page_scroller}>
      <div className={styles.page}>
        <Sidebar filters={filters} />
        <section className={styles.content}>
          <div className={styles.content_container}>
            <Switch>
              <Route
                path={`${getPublicPath()}:loadName`}
                render={routeProps => {
                  const { loadName } = routeProps.match.params
                  const definition = getDefinition(loadName)

                  return definition ? (
                    <LabwareDetails definition={definition} />
                  ) : (
                    <NoResults />
                  )
                }}
              />
              <Route render={() => <LabwareList filters={filters} />} />
            </Switch>
          </div>
        </section>
      </div>
    </div>
  )
}
