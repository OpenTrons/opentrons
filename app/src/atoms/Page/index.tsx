import * as React from 'react'

import styles from './styles.css'
import { TitleBar } from '@opentrons/components'

import type { TitleBarProps } from '@opentrons/components'

export interface PageProps {
  titleBarProps?: TitleBarProps
  children: React.ReactNode
}

export function Page(props: PageProps): JSX.Element {
  const { titleBarProps, children } = props

  return (
    <main className={styles.page}>
      {titleBarProps && (
        <TitleBar {...titleBarProps} className={styles.fixed_title_bar} />
      )}
      {children}
    </main>
  )
}
