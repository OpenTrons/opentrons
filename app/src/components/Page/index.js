// @flow
import * as React from 'react'

import styles from './styles.css'
import { TitleBar, type TitleBarProps } from '@opentrons/components'
import PageWrapper from './PageWrapper'
import RefreshWrapper from './RefreshWrapper'

type Props = {
  titleBarProps?: TitleBarProps,
  children: React.Node,
}

export default function Page(props: Props) {
  const { titleBarProps, children } = props
  return (
    <main className={styles.task}>
      {titleBarProps && (
        <TitleBar {...titleBarProps} className={styles.fixed_title_bar} />
      )}
      {children}
    </main>
  )
}

export { PageWrapper, RefreshWrapper }
