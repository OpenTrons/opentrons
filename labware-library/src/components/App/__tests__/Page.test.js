// @flow
// app tests
import * as React from 'react'
import { shallow } from 'enzyme'

import Page from '../Page'

jest.mock('../../../definitions')

describe('Page', () => {
  test('component renders', () => {
    const tree = shallow(<Page location={({ search: '' }: any)} />)

    expect(tree).toMatchSnapshot()
  })
})
