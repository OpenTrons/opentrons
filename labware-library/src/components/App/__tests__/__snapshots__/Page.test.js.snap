// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page component renders 1`] = `
<div
  className="page_scroller"
>
  <div
    className="page"
  >
    <Sidebar
      filters={
        Object {
          "category": "all",
          "manufacturer": "all",
        }
      }
    />
    <section
      className="content"
    >
      <div
        className="content_container"
      >
        <Switch>
          <Route
            path="/:loadName"
            render={[Function]}
          />
          <Route
            render={[Function]}
          />
        </Switch>
      </div>
    </section>
  </div>
</div>
`;
